src/main/java/com/minecolonies/coremod/blocks/BlockDecorationController.java

```diff
@@ -93,21 +99,64 @@ public ResourceLocation getRegistryName()
     }
 
     @Override
-    public VoxelShape getShape(final BlockState state, final BlockGetter worldIn, final BlockPos pos, final CollisionContext context)
+    public VoxelShape getShape(final BlockState decoController, final BlockGetter level, final BlockPos pos, final CollisionContext context)
     {
-        Direction Direction = state.getValue(FACING);
-        switch (Direction)
+        final Direction direction = decoController.getValue(BlockDecorationController.FACING);
+        final BlockPos offsetPos = pos.relative(direction);
+        final BlockState state = level.getBlockState(offsetPos);
+        final VoxelShape shape = state.getShape(level, offsetPos);
+        if (shape.isEmpty() || Block.isShapeFullBlock(shape))
         {
-            case EAST:
-                return AABB_EAST;
-            case WEST:
-                return AABB_WEST;
-            case SOUTH:
-                return AABB_SOUTH;
-            case NORTH:
-            default:
-                return AABB_NORTH;
+            return switch (direction)
+                     {
+                         case EAST -> AABB_EAST;
+                         case WEST -> AABB_WEST;
+                         case SOUTH -> AABB_SOUTH;
+                         case NORTH -> AABB_NORTH;
+                         case UP -> AABB_UP;
+                         case DOWN -> AABB_DOWN;
+                     };
         }
+
+        final Vec3 translateVec;
+        switch (direction)
+        {
+            case UP ->
+            {
+                translateVec = new Vec3(0, shape.min(Direction.Axis.Y), 0);
+            }
+            case DOWN ->
+            {
+                translateVec = new Vec3(0, shape.max(Direction.Axis.Y) - 1, 0);
+            }
+            case NORTH ->
+            {
+                translateVec = new Vec3(0, 0, shape.max(Direction.Axis.Z) - 1);
+            }
+            case SOUTH ->
+            {
+                translateVec = new Vec3(0, 0, shape.min(Direction.Axis.Z));
+            }
+            case EAST ->
+            {
+                translateVec = new Vec3(shape.min(Direction.Axis.X), 0, 0);
+            }
+            case WEST ->
+            {
+                translateVec = new Vec3(shape.max(Direction.Axis.X) - 1, 0, 0);
+            }
+            default -> translateVec = new Vec3(0, 0, 0);
+        }
```

Comment:
This is an interesting combination of switch statement and switch expression syntax
I either would have expected this to be a switch statement, or the translateVec assignment to be out of the switch expression
```java
final Vec3 translateVec = switch(direction)
{
    case UP -> new Vec3(...)
    case DOWN -> new Vec3(...)
    // etc.
}
```